python/lsst/pipe/base/pipelineIR.py

```diff
@@ -41,6 +50,36 @@ class KeepInstrument:
     pass
 
 
+def standardize_uri_in_pipeline(path: str, directory: Optional[ButlerURI]) -> ButlerURI:
+    """Standardize a URI inside a pipeline description by expanding environment
+    variables and interpreting (or rejecting) relative paths.
+
+    Parameters
+    ----------
+    path : `str`
+        Path to standardize.
+    directory : `ButlerURI` or `None`
+        Directory URI that relative paths should be considered relative to.
+
+    Returns
+    -------
+    uri : `ButlerURI`
+        Standardized URI.
+
+    Raises
+    ------
+    RuntimeError
+        Raised if ``path`` is relative but ``directory`` is `None`.
+    """
+    uri = ButlerURI(os.path.expandvars(path), forceAbsolute=False)
```

Comment:
`expandvars` does now happen inside ButleRURI for schemeless URIs so this is no longer needed (when the butler Config code was written that wasn't true).